modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java

```diff
@@ -18,429 +18,630 @@
+                                     inProgressOperations);
+                }
+            }
+        }
+    }
+
+    public void updateDeviceOperations(Status status, SyncmlDocument syncmlDocument,
+                                       DeviceIdentifier deviceIdentifier)
+            throws OperationManagementException, DeviceManagementException {
+
+        inProgressOperations = SyncmlUtils.getDeviceManagementService()
+                .getOperationsByDeviceAndStatus(deviceIdentifier, Operation.Status.IN_PROGRESS);
+        if (status.getData().equals(Constants.SyncMLResponseCodes.ACCEPTED) || status.getData().equals
+                (Constants.SyncMLResponseCodes.ACCEPTED_FOR_PROCESSING)) {
+            for (int x = 0; x < inProgressOperations.size(); x++) {
+                Operation operation = inProgressOperations.get(x);
+                if (operation.getId() == status.getCommandReference()) {
+                    operation.setStatus(Operation.Status.COMPLETED);
+                    operation.setOperationResponse("true");
+                }
+            }
+            updateOperations(syncmlDocument.getHeader().getSource().getLocURI(), inProgressOperations);
+        } else if (status.getData().equals(Constants.SyncMLResponseCodes.PIN_NOTFOUND)) {
+            for (int x = 0; x < inProgressOperations.size(); x++) {
+                Operation operation = inProgressOperations.get(x);
+                if (operation.getId() == status.getCommandReference() && inProgressOperations.get(x).
+                        getCode().equals(String.valueOf(OperationCode.Command.DEVICE_LOCK))) {
+                    operation.setStatus(Operation.Status.ERROR);
+                    operation.setOperationResponse("false");
+                    updateOperations(syncmlDocument.getHeader().getSource().getLocURI(), inProgressOperations);
+                    try {
+                        NotificationManagementService service =
+                                WindowsAPIUtils.getNotificationManagementService();
+                        Notification lockResetNotification = new Notification();
+                        lockResetNotification.setOperationId(status.getCommandReference());
+                        lockResetNotification.setStatus(String.valueOf(Notification.Status.NEW));
+                        lockResetNotification.setDeviceIdentifier(deviceIdentifier);
+                        lockResetNotification.setDescription(
+                                Constants.SyncMLResponseCodes.LOCKRESET_NOTIFICATION);
+                        service.addNotification(lockResetNotification);
+                    } catch (NotificationManagementException e) {
+                        String msg = "Failure occurred in getting notification service";
+                        log.error(msg);
+                    }
+                }
+            }
+        }
+    }
+
+    public List<Operation> getDeviceInfo() {
```

Comment:
No need to construct this object over and over again, AFAIU. Can we initialize this in a static block up in the class declaration and re-use?